sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -618,3 +619,44 @@ case class SQLKeywords() extends LeafExpression with Generator with CodegenFallb
 
   override def prettyName: String = "sql_keywords"
 }
+
+@ExpressionDescription(
+  usage = """_FUNC_() - Get all of the Spark SQL string collations""",
+  examples = """
+    Examples:
+      > SELECT * FROM _FUNC_() WHERE COLLATION_NAME = 'UTF8_BINARY';
+       SYSTEM  BUILTIN  UTF8_BINARY NULL  NULL  ACCENT_SENSITIVE  CASE_SENSITIVE  NO_PAD  NULL
+  """,
+  since = "4.0.0",
+  group = "generator_funcs")
+case class AllCollations() extends LeafExpression with Generator with CodegenFallback {
+  override def elementSchema: StructType = new StructType()
+    .add("COLLATION_CATALOG", StringType, nullable = false)
+    .add("COLLATION_SCHEMA", StringType, nullable = false)
+    .add("COLLATION_NAME", StringType, nullable = false)
```

Comment:
do we really need the `COLLATION` prefix? Can it just be `CATALOG`, `SCHEMA` and `NAME`?

Reply:
I agree with it because `sql_keywords` doesn't seem to have a prefix either.
<img width="771" alt="image" src="https://github.com/user-attachments/assets/f7c30133-5686-4fb1-8a97-63b7fd372138">

Reply:
Also, I have found an issue. Do we need to register it as a `SQL built-in function`? I noticed that some functions in `FunctionRegistry` are both `TVF` and `SQL built-in function`, eg:
https://github.com/apache/spark/blob/9fc58aa4c0753ef42e0172d73e13b45a00a730f9/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L370-L371
https://github.com/apache/spark/blob/9fc58aa4c0753ef42e0172d73e13b45a00a730f9/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L1155-L1156



so we can use them as follows:
```sql
SELECT all_collations();
```

Reply:
Using generator functions in SELECT is not recommended. We only keep them for backward compatibility.

Reply:
Thank you, I see.